lib/elastic_apm/transport/worker.rb

```diff
@@ -58,7 +58,10 @@ def process(resource)
 
       def serialize_and_filter(resource)
         serialized = serializers.serialize(resource)
-        @filters.apply!(serialized)
+
+        # if a filter returns nil, it means skip the event
+        return nil if @filters.apply!(serialized) == Filters::SKIP
```

Comment:
Reading `filters.apply!(serialized)` I would expect the `filters` to be changed in place and not `serialized`. (I know that you didn't introduce this right now).  Would it help if you derive the serialized events from a common base class offering a method for filtering, e.g. `serialized_event.apply!(filters)`. Afterwards, you could check if `serialized_event.body` is nil and move on or not.

Reply:
Could you write a short (but slightly longer) example?

Reply:
Ah, now I get it. Renaming the method `apply_to` would solve it too 😉